tests/cli/test_cli.py

```diff
@@ -0,0 +1,65 @@
+import inspect
+import json
+import pathlib
+
+from typer.testing import CliRunner
+
+from hamilton.cli.__main__ import cli
+
+from tests.resources import example_module
+
+runner = CliRunner()
+
+
+def test_cli_build_success():
+    module_path = inspect.getfile(example_module)
+    inputs = ["--json-out", "build", module_path]
+
+    result = runner.invoke(cli, inputs)
+    message = json.loads(result.stdout)
+
+    assert result.exit_code == 0
+    assert message["success"] is True
+
+
+def test_cli_view_default(tmp_path: pathlib.Path):
+    module_path = inspect.getfile(example_module)
+    inputs = ["--json-out", "view", module_path]
+    # output_file_path = tmp_path / "dag.png"  # default file name
+
+    result = runner.invoke(cli, inputs)
+    message = json.loads(result.stdout)
+
+    assert result.exit_code == 0
+    assert message["success"] is True
+    # TODO difficult to assert the output file path from test
```

Comment:
Why is this?

Reply:
My manual verification of the `view --output` option works. 

I believe the `typer.testing.CliRunner` is instantiated at the pytest top-level and isn't instantiated in the relative `tmp_path` that is scoped to a single test function.

Could be fixed by passing fixtures, but probably requires more state management across tests...